examples/protocol_specification_ex/tac.yaml

```diff
@@ -0,0 +1,45 @@
+name: tac
+author: fetchai
+version: 0.1.0
+license: Apache 2.0
+description: 'A protocol for participating in a TAC.'
+speech_acts:
+  register:
+    agent_name: str
+  unregister: {}
+  transaction:
+    tx_id: str
+    tx_sender_addr: str
+    tx_counterparty_addr: str
+    amount_by_currency_id: dict
+    tx_sender_fee: int
+    tx_counterparty_fee: int
+    quantities_by_good_id: dict
+    tx_nonce: int
+    tx_sender_signature: bytes
+    tx_counterparty_signature: bytes
+  get_state_update: {}
+  cancelled: {}
+  game_data:
+    amount_by_currency_id: dict
+    exchange_params_by_currency_id: dict
+    quantities_by_good_id: dict
+    utility_params_by_good_id: dict
+    tx_fee: int
+    agent_addr_to_name: dict
+    good_id_to_name: dict
+    version_id: str
+  transaction_confirmation:
+    tx_id: str
+    amount_by_currency_id: dict
+    quantities_by_good_id: dict
+  tac_error:
+    error_code: ErrorCode
+    info: dict
+
+# transaction -> amount_by_currency_id is a dictionary. Declare each key, value in 'amount_by_currency_id' is of type str int respectively.
+# similarly for quantities_by_good_id, amount_by_currency_id, exchange_params_by_currency_id, quantities_by_good_id, utility_params_by_good_id, good_id_to_name,
+# currently you check whether keys in the above are unique (line 225). Is that necessary give it is dictionary?
```

Comment:
No keys are by design unique, so no need to check.

Reply:
You mean I remove lines 225, 233, 269, 273 (where the checks for uniqueness happen) from message.py of tac protocol?

Reply:
You don't need to change existing protocols. We will change them once the generator is finished in its first iteration.

Reply:
ok